### ISSUE TITLE
No Bech32 support

### ISSUE THREAD

**Issue by bjarnemagnussen** [ofek/bit#82]
Currently Bit has all the logic to support generating (receiving) Bech32 addresses and this could be added relatively easily. 

The reason Bit does not yet support it is because two out of the three used WebAPIs don't support Bech32. Those are blockchain.info and Insight by BitPay. Smartbit on the other hand does support Bech32.

Two new WebAPIs exist that support Bech32. Those are blockstream.info and blockchair, both of which provide a rich enough API that can be integrated with Bit.

It has been two years now since Segwit was activated and I have been considering to suggest that we completely remove blockchain.info and insight from our NetworkAPI and instead add blockstream.info and blockchair in addition to keeping Smartbit. 

One problem of removing specific APIs could be that some have been been using them in Bit with authentication to circumvent the request limitations. However, since that is not officially supported it requires custom patching of Bit and therefore removing support for specific APIs should not result in breaking some people's code by surprise?

What do you all think, should we go ahead and remove those APIs that still don't support Bech32; or work around them by distinguishing between Bech32 and non-Bech32 supporting APIs but which would increase the burden of maintaining the code dramatically in my opinion and is not worth it.

**Comment by teran-mckinney** [comment]
Personally, I think it would be ideal to keep them but maybe have a features = [] that would say the new ones support bech32 and the old ones don't. Then use whichever ones as needed in code.

There's lots of issues with these APIs all the time. And a lot of them block Tor, so only one or two of them might be working at a time for Tor users. Being able to rotate and get a transaction out is nice.

**Comment by bjarnemagnussen** [comment]
Are those issues mostly related to sending out transactions, or is it equally problematic to receive transaction information over Tor?

Blockstream.info has an onion site for its API and so it may be important to also think about a Tor parameter in the `NetworkAPI` to connect to the onion site(s) directly.

**Comment by teran-mckinney** [comment]
Both are issues, any API on Cloudflare that hasn't allowed the T1 region tends to block all requests.

Blockchain.info also has a .onion which is kind of a pain, if you hit the .info over Tor it will refuse the request and say you have to use the .onion. I agree with you, would be nice to have a Tor/.onion mode.

**Comment by ofek** [comment]
When we support Tor we'll need `requests[socks]` here:

https://github.com/ofek/bit/blob/991d7f17bea00b718de0c5950a4637ef98fc16e3/setup.py#L44

**Comment by teran-mckinney** [comment]
Yes. Not only that, bad a lot of the brokens are version. I've been doing this lately:

https://github.com/sporestack/sporestack-python/blob/master/setup.py#L36

Had all kinds of troubles until that one.

Here's how I've been handling it:

https://github.com/sporestack/sporestack-python/blob/master/sporestackv2/api_client.py#L19-L72

**Comment by bjarnemagnussen** [comment]
Thanks for the info!

**Comment by Th0rgal** [comment]
Do you plan to add support for native segwit (bech32)?

**Comment by j1warren** [comment]
Any news up on this?
